libnymea-core/hardware/network/mqtt/mqttproviderimplementation.cpp

```diff
@@ -205,15 +205,17 @@ void MqttProviderImplementation::setEnabled(bool enabled)
 void MqttProviderImplementation::onClientConnected(const QString &clientId)
 {
     if (m_createdChannels.contains(clientId)) {
-        MqttChannel* channel = m_createdChannels.value(clientId);
+        MqttChannelImplementation *channel = qobject_cast<MqttChannelImplementation *>(m_createdChannels.value(clientId));
+        channel->setConnected(true);
         emit channel->clientConnected(channel);
```

Comment:
I suppose at this point it would be nicer if you'd just call channel->setConnected(true) here and emit the clientConnected(this) from inside the channel?
Same with the disconnected below